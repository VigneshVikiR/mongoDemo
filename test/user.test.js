const myFunc = jest
  .fn()
  .mockImplementationOnce(() => 'first call')
  .mockImplementationOnce(() => 'second call');

const storeData = require('../handler/user').storeUserData;
const model = require('../model/user');

jest.mock('model');

describe('test my function', () => {
  model.storeUserData.mockResolvedValue({
    body: {
      id: '1',
      name: 'eee',
      emailId: '233',
      mobNo: 3232332,
    }
  })
  it('store user data', async done => {
    const request = {
      body: {
        id: '1',
        name: 'eee',
        emailId: '233',
        mobNo: 3232332,
      }
    };
    const response = {
      send: output => {
        done('User data created successfully')
      }
    };
    const next = () => {
    };
    try {
      await storeData(request, response, next)
    } catch (e) {
      console.log(e)
    }
  })
});
