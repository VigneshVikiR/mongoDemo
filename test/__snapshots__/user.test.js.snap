// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test my function first suite 1`] = `
[MockFunction] {
  "calls": Array [
    Array [],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": "first call",
    },
  ],
}
`;
